test/com/facebook/buck/jvm/groovy/GroovyLibraryIntegrationTest.java

```diff
@@ -66,15 +64,38 @@ public void shouldCompileLibraryWithDependencyOnAnother() throws Exception {
 
   @Test
   public void shouldFailToCompileInvalidGroovyClass() throws Exception {
-    assumeTrue(System.getenv("GROOVY_HOME") != null);
+    ProjectWorkspace.ProcessResult buildResult =
+        workspace.runBuckCommand("build", "//com/example/bad:fail");
+    buildResult.assertFailure();
 
-    workspace = TestDataHelper.createProjectWorkspaceForScenario(
-        this, "groovy_library_description", tmp);
-    workspace.setUp();
+    workspace.verify();
+  }
 
-    // Run `buck build`.
+  @Test
+  public void shouldCrossCompileWithJava() throws Exception {
     ProjectWorkspace.ProcessResult buildResult =
-        workspace.runBuckCommand("build", "//com/example/bad:fail");
+        workspace.runBuckCommand("build", "//com/example/xcompile:xcompile");
+    buildResult.assertSuccess();
+
+    workspace.verify();
+  }
+
+  @Test
+  public void javaOptionsArePassedThroughToTheJavacCompiler() throws Exception {
+    // code requires java 7, source set to java 6
```

Comment:
nit: please use complete sentences, which means you need capital letters and punctuation at the end.